### DISCUSSION TITLE
Ussage in Ci/Cd

### DISCUSSION THREAD

**Discussion by JDubbTX** [IBM/ibmi-tobi#234] [General]
I am trying to use Bob in a CiCd pipeline with gitlab, which has its challenges - but have the following question.

For the build portion of the ci/cd pipeline, is it possible to have bob sense what objects need compiling in the application library, but build the objects to a different package library that I create on the fly, just prior to running makei?  

Idea is to package built objects into a library, then create a savf from that library, download as an artifact to the gitlab runner which will deploy the artifact to artifactory.  I don't want to deploy the entire application, just the newly built objects.  A later job in the pipeline will pick up the package, deploy it to the target lpar, and extract the contents of the savf to the original application library.  

Is this possible? How would I tell bob to build to a different library?

**Comment by edmundreinhardt** [comment]
Bob is intended to be used in a CI/CD pipeline.
The iproj.json should specify the objlib using a variable.  Based on the variable value BOB will build into different libraries. 
So before invoking BOB for the build, simply set the environment variable to the desired library.

**Reply by edmundreinhardt** [reply]
In the context of a CI/CD pipeline, the expectation is that all files are cloned into a fresh directory and built using BOB.

**Comment by edmundreinhardt** [comment]
Only developer builds are optimized to build the subset of source that has changed
